Start the work for the options page

aguspe opened the pull request:
## **User description**
### Description
This PR aims to start improving the example for browser options in the Selenium documentation for the ruby bindings

Currently, most of the examples are lacking and some of the code needs to be moved

Examples:

![Screenshot 2024-04-15 at 18 04 05](https://github.com/SeleniumHQ/seleniumhq.github.io/assets/33221555/c690bc26-4f2f-47b5-b331-7e42fc974a61)


### Motivation and Context
For both newcomers and experienced Selenium users is important to have up-to-date documentation that they can rely on

This update will be split into multiple PRs to not make one big PR and to start updating the documentation right now

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Change to the site (I have double-checked the Netlify deployment, and my changes look good)
- [ ] Code example added (and I also added the example to all translated languages)
- [ ] Improved translation
- [ ] Added new translation (and I also added a notice to each document missing translation)

### Checklist
- [ ] I have read the [**contributing**](https://selenium.dev/documentation/en/contributing/) document.
- [ ] I have used [hugo](https://gohugo.io/) to render the site/docs locally and I am sure it works.


___

## **Type**
enhancement, documentation


___

## **Description**
- Added and updated Ruby examples for Chrome driver options, including page load strategies and setting remote capabilities.
- Updated WebDriver options documentation across multiple languages (English, Japanese, Portuguese, Chinese) with new Ruby examples.
- Enhanced the upgrade to Selenium 4 documentation with updated Ruby code examples.



___



## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>options_spec.rb</strong><dd><code>Enhance Ruby Examples for Browser Options</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/ruby/spec/drivers/options_spec.rb

<li>Added examples for Chrome driver options including page load <br>strategies and remote capabilities.<br> <li> Introduced a new test case for setting remote capabilities with <br>platform name, browser version, and cloud options.<br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1668/files#diff-14ec885660d66546724e448a9870b3afdd72ec096c7d4777539e3e4dfb65c39c">+24/-11</a>&nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>options.en.md</strong><dd><code>Update Ruby Examples in WebDriver Options Documentation</code>&nbsp; &nbsp; </dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/drivers/options.en.md

<li>Updated Ruby code blocks to reference the new examples in <br><code>options_spec.rb</code>.<br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1668/files#diff-62bab9f28d27a7f462881a28eca384a0e534a730f849248f396d171820bc748e">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>options.ja.md</strong><dd><code>Update Japanese WebDriver Options Documentation with Ruby Examples</code></dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/drivers/options.ja.md

<li>Updated Ruby code blocks to include new examples from <code>options_spec.rb</code>.<br> <br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1668/files#diff-2ed475220bde1368cfa3883d71db179409ddaf9ccbaea847ed3f6ffe816321e4">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>options.pt-br.md</strong><dd><code>Enhance Portuguese WebDriver Options Documentation with Ruby Examples</code></dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/drivers/options.pt-br.md

<li>Added new Ruby examples in the WebDriver options documentation for <br>Portuguese readers.<br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1668/files#diff-6eaa598a13e9d64c6afbd54b84aef470520557f08c4d183ffb28ceb9ed04d0bb">+23/-2</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>options.zh-cn.md</strong><dd><code>Update Chinese WebDriver Options Documentation with Ruby Examples</code></dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/drivers/options.zh-cn.md

<li>Included updated Ruby examples in the Chinese WebDriver options <br>documentation.<br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1668/files#diff-9b23b749355db8154bfcc4a1d70f809b7d4e91a7b3044f9ac74a6c1a005873d6">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>upgrade_to_selenium_4.en.md</strong><dd><code>Update Upgrade to Selenium 4 Documentation with New Ruby Examples</code></dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/troubleshooting/upgrade_to_selenium_4.en.md

<li>Replaced outdated Ruby code with a reference to the new <br><code>options_spec.rb</code> examples for upgrading to Selenium 4.<br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1668/files#diff-f2eb17a1ff8de1d066986be0ea8c3f5158b6b85c0004071e832a508ceb27eb0a">+1/-8</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

aguspe committed:
Start the work for the options page

aguspe committed:
Merge branch 'trunk' into add_browser_options_ruby_examples

aguspe committed:
Move upgrade to selenium examples

aguspe committed:
Update platform name

aguspe committed:
Merge branch 'trunk' into add_browser_options_ruby_examples

aguspe committed:
Update platform name and platform version

aguspe committed:
Merge branch 'trunk' into add_browser_options_ruby_examples

aguspe commented:
@diemol my Portuguese is not great but I noticed that the Portuguese translation was the only one that was lacking the browserName examples

Also, I wanted to update all the ruby examples but I did not wanted to give you or others reviewers too much work in one single PR

diemol commented on examples/ruby/spec/drivers/options_spec.rb:
```diff
@@ -6,31 +6,44 @@
-  end
+    end
+
+    it 'sets remote capabilities' do
+      options = Selenium::WebDriver::Options.firefox
+      options.platform_name = 'Windows 10'
+      options.browser_version = 'latest'
+      cloud_options = {}
+      cloud_options[:build] = my_test_build
+      cloud_options[:name] = my_test_name
+      options.add_option('cloud:options', cloud_options)
+      driver = Selenium::WebDriver.for :remote, url: cloud_url, capabilities: options
+      driver.get('https://www.google.com')
+      driver.quit
+    end
```
I think we cannot execute this code. It would error, right? I believe, `cloud_url` is not defined. 

This code can probably stay in the migration guide.

aguspe replied:
> I think we cannot execute this code. It would error, right? I believe, `cloud_url` is not defined.
> 
> This code can probably stay in the migration guide.

You are right, but I added the option to skip it automatically in rspec during execution with an explanation why, in that way we can move the code, and reference it in multiple places without executing the actual test

Let me know if you like this solution, if not I will move the code back where it was

diemol commented:
> @diemol my Portuguese is not great but I noticed that the Portuguese translation was the only one that was lacking the browserName examples
> 
> Also, I wanted to update all the ruby examples but I did not wanted to give you or others reviewers too much work in one single PR

Yes, this was probably overlooked. Feel free to change that, please. Thanks!

aguspe commented:
> > @diemol my Portuguese is not great but I noticed that the Portuguese translation was the only one that was lacking the browserName examples
> 
> > 
> 
> > Also, I wanted to update all the ruby examples but I did not wanted to give you or others reviewers too much work in one single PR
> 
> 
> 
> Yes, this was probably overlooked. Feel free to change that, please. Thanks! 

That's already changed 😊 thank you for the quick response!

diemol commented:
> > > @diemol my Portuguese is not great but I noticed that the Portuguese translation was the only one that was lacking the browserName examples
> > 
> > 
> > > 
> > 
> > 
> > > Also, I wanted to update all the ruby examples but I did not wanted to give you or others reviewers too much work in one single PR
> > 
> > 
> > Yes, this was probably overlooked. Feel free to change that, please. Thanks!
> 
> That's already changed 😊 thank you for the quick response!

I don't see the changes. And there is an open conversation that needs to be resolved. Thank you!

luisfcorreia commented:
Just a comment, were you suggesting translating code examples to PT or the surrounding text?

(I got confused)

aguspe commented:
> Just a comment, were you suggesting translating code examples to PT or the surrounding text?
> 
> (I got confused)

Hi Luis, I noticed that the code examples for browser name were missing in the Portuguese translations as it is now

![Screenshot 2024-04-16 at 16 57 08](https://github.com/SeleniumHQ/seleniumhq.github.io/assets/33221555/e9c63828-3a10-4d40-a41a-67f92eaf2024)

Compared to the English version:

![Screenshot 2024-04-16 at 16 57 26](https://github.com/SeleniumHQ/seleniumhq.github.io/assets/33221555/a99ce2e8-ef79-430b-8e86-6c02c1ed41db)


So I was wondering if it was on purpose or if the examples were missing :)

Docs URL: https://www.selenium.dev/documentation/webdriver/drivers/options/

aguspe committed:
Merge branch 'trunk' into add_browser_options_ruby_examples

aguspe commented:
Perfect I will address it tonight, thank you!

aguspe committed:
Merge branch 'trunk' into add_browser_options_ruby_examples

aguspe committed:
Add skip method

aguspe committed:
Merge remote-tracking branch 'origin/add_browser_options_ruby_examples' into add_browser_options_ruby_examples

aguspe commented:
@diemol  Regarding your request for changes, you are absolutely right, the test will not execute

So I implemented an alternative solution, I added the option to skip the test automatically in rspec during execution with an explanation why, in that way we can move the code, and reference it in multiple places without executing the actual test

Let me know if you like this solution, if not I will move the code back where it was

diemol reviewed (approved):
That works, thank you!
